### DISCUSSION TITLE
How can I use dynamic variable in field validation?

### DISCUSSION THREAD

**Discussion by khanetor** [directus/directus#17836] [Archive]
I have a field of type DATETIME, and I want that when creating, the datetime should be in the future, not in the past. So I want to create a validation filter with a `$NOW` variable.

This is my attempt 
![image](https://user-images.githubusercontent.com/3459416/226057604-fa8714fa-d420-401a-8c34-fc46e485326d.png)

I have tried using curly braces, and double quotes, with I still have the same error `{
  "message": "date must have a valid date format or reference",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}`
Which is unrelated to the validation rule.

To save effort on this question, let's find out:
1. Is it possible to use dynamic variables in field validation at all? And if yes,
2. How can I specify this dynamic variable in the UI? Or if it is not possible in the UI,
3. Can I create field validation with JSON?

**Comment by Wendenburg** [comment]
Same here, something like **Greater than $NOW(+2 days)** is simply not working and creating the error mentioned above.

**Comment by simonflick** [comment]
I need something similar:
Given 2 fields `date_begin` and `date_end` I want to force `date_end` >= `date_begin`.

**Reply by khanetor** [reply]
> I need something similar: Given 2 fields `date_begin` and `date_end` I want to force `date_end` >= `date_begin`.

I think you can do this with condition, rather than validation. Validation is for validating a field, independently from other fields. Condition is for validating a field, in relative to other fields.

**Reply by simonflick** [reply]
Thanks, conditions didn't work when I tried to use them but I will give it another shot.

**Comment by vanekj** [comment]
It seems that it still does not work. I am running Directus `10.8.3` and I am still unable to use dynamic variables in field validations. Anyone found solution to this? 🙏🏻

**Reply by khanetor** [reply]
I think we need to make feature request before this is noticed.

**Comment by alepietrobon** [comment]
hi, i'm in the same boat here. I think this is flagged here:  https://github.com/directus/directus/issues/14759

Bug is still not closed
